### PR TITLE
Add `vpc.v1.subnet` resource

### PR DESCRIPTION
**Pull request by outcatcher**
Add subnet methods to `vpc.v1.proxy`

Add acceptance tests for `vpc.v1.subnet`

Add `vpc.v1.vpc` and `vpc.v1.subnet` documentation

### CODE REVIEW

File: doc/source/sdk/resources/vpc/v1/subnet.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
+otcextensions.sdk.vpc.v1.subnet
+================================
+
+.. automodule:: otcextensions.sdk.vpc.v1.subnet
+
+The VPC Class
```
Comment (lego963) [inline doc/source/sdk/resources/vpc/v1/subnet.rst]: VPC class?

File: otcextensions/sdk/vpc/v1/subnet.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+# under the License.
+from openstack import resource
+
+
+class ExtraDHCPOpt(resource.Resource):
+    #: Specifies the NTP server address configured for the subnet.
+    opt_value = resource.Body('opt_value')
+    #: Specifies the NTP server address name configured for the subnet.
+    #: Currently, the value can only be set to ntp.
+    opt_name = resource.Body('opt_name', default='ntp')
```
Comment (lego963) [inline otcextensions/sdk/vpc/v1/subnet.py:20]: maybe both of them should have defaults

**Review by lego963 — CHANGES_REQUESTED** [review]

File: otcextensions/sdk/vpc/v1/_proxy.py [hunk 3]
```diff
@@ -249,6 +254,122 @@ def update_vpc(self, vpc, **attrs):
...
+
+        attrs.pop('base_path', None)
+        base_path = _subnet.vpc_subnet_base_path(vpc_id)
+
+        return self._update(_subnet.Subnet,
+                            subnet,
+                            base_path=base_path,
+                            **attrs)
+
+    def _delete(self, resource_type, value, ignore_missing=True, **attrs):
```
Comment (anton-sidelnikov) [inline otcextensions/sdk/vpc/v1/_proxy.py:334]: better move it to resource

File: otcextensions/tests/functional/sdk/vpc/v1/test_subnet.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+        self.client = self.conn.vpc
+        attrs = {
+            'name': 'test-vpc-' + self.seed,
+            'cidr': '192.168.0.0/16'
+        }
+        self.vpc = self.conn.vpc.create_vpc(**attrs)
+        self.addCleanup(self.conn.vpc.delete_vpc, self.vpc)
+
+    def test_initialize(self):
```
Comment (anton-sidelnikov) [inline otcextensions/tests/functional/sdk/vpc/v1/test_subnet.py]: i donnu why, but this test always in separate file

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: otcextensions/sdk/vpc/v1/subnet.py (hunk 2 above)
Comment (outcatcher) [reply otcextensions/sdk/vpc/v1/subnet.py:20]: The `opt_value` is not mandatory and has the following limitations:
```
The option ntp for opt_name indicates the NTP server configured for the subnet. Currently, only IPv4 addresses are supported. A maximum of four IP addresses can be configured, and each address must be unique. Multiple IP addresses must be separated using commas (,). The option null for opt_name indicates that no NTP server is configured for the subnet. The parameter value cannot be an empty string.
```
And which IP address value can be used in this case?

**Review by anton-sidelnikov — APPROVED** [review]

**Review by lego963 — APPROVED** [review]

**Pull request merged**